pkg/transformers/json/utils.go

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package json
+
+import (
+	"math"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/mainflux/mainflux/pkg/errors"
+)
+
+func parseTimestamp(format string, timestamp interface{}, location string) (time.Time, error) {
+	switch format {
+	case "unix", "unix_ms", "unix_us", "unix_ns":
+		return parseUnix(format, timestamp)
+	default:
+		if location == "" {
+			location = "UTC"
+		}
+		return parseTime(format, timestamp, location)
+	}
+}
+
+func parseUnix(format string, timestamp interface{}) (time.Time, error) {
+	integer, fractional, err := parseComponents(timestamp)
+	if err != nil {
+		return time.Unix(0, 0), err
+	}
+
+	switch strings.ToLower(format) {
+	case "unix":
+		return time.Unix(integer, fractional).UTC(), nil
+	case "unix_ms":
+		return time.Unix(0, integer*1e6).UTC(), nil
+	case "unix_us":
+		return time.Unix(0, integer*1e3).UTC(), nil
+	case "unix_ns":
+		return time.Unix(0, integer).UTC(), nil
+	default:
+		return time.Unix(0, 0), errors.New("unsupported type")
+	}
+}
+
+func parseComponents(timestamp interface{}) (int64, int64, error) {
+	switch ts := timestamp.(type) {
+	case string:
+		parts := strings.SplitN(ts, ".", 2)
+		if len(parts) == 2 {
+			return parseUnixTimeComponents(parts[0], parts[1])
+		}
+
+		parts = strings.SplitN(ts, ",", 2)
+		if len(parts) == 2 {
+			return parseUnixTimeComponents(parts[0], parts[1])
+		}
+
+		integer, err := strconv.ParseInt(ts, 10, 64)
+		if err != nil {
+			return 0, 0, err
+		}
+		return integer, 0, nil
+	case int8:
+		return int64(ts), 0, nil
+	case int16:
+		return int64(ts), 0, nil
+	case int32:
+		return int64(ts), 0, nil
+	case int64:
+		return ts, 0, nil
+	case uint8:
+		return int64(ts), 0, nil
+	case uint16:
+		return int64(ts), 0, nil
+	case uint32:
+		return int64(ts), 0, nil
+	case uint64:
+		return int64(ts), 0, nil
```

Comment:
Join to a single case: 
```go

case int8, int16, int32...:
    return int64(ts), 0, nil
```

Reply:
This won't work as a single case since we need to determine the exact type.